src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -245,11 +291,19 @@ void HexagonGame::draw()
             }
 
             // Draw player layers
-            for(std::size_t k = j * numPlayerTris; k < (j + 1) * numPlayerTris;
-                ++k)
+            for(std::size_t k = jAdj * numPlayerTris;
+                k < (jAdj + 1) * numPlayerTris; ++k)
             {
-                playerTris3D[k].position += newPos;
-                playerTris3D[k].color = overrideColor;
+                if(renderingAbove)
+                {
+                    playerTris3DTop[k].position += newPos;
+                    playerTris3DTop[k].color = overrideColor;
+                }
+                else
+                {
+                    playerTris3D[k].position += newPos;
+                    playerTris3D[k].color = overrideColor;
+                }
```

Comment:
Pull the `if` statement outside of the loop, so that we have `if (renderAbove) { for{...} } else { for{...} }`.